dhcpv6/server6/server.go

```diff
@@ -122,10 +127,29 @@ func NewServer(addr *net.UDPAddr, handler Handler, opt ...ServerOpt) (*Server, e
 	}
 
 	if s.conn == nil {
+		// no connection provided by the user, create a new one
 		conn, err := net.ListenUDP("udp6", addr)
 		if err != nil {
 			return nil, err
 		}
+		// join multicast group on the specified interface
+		var iface *net.Interface
+		if ifname == "" {
+			iface = nil
+		} else {
+			iface, err = net.InterfaceByName(ifname)
+			if err != nil {
+				return nil, err
+			}
+		}
+		group := net.UDPAddr{
+			IP:   dhcpv6.AllDHCPRelayAgentsAndServers,
+			Port: dhcpv6.DefaultServerPort,
+		}
+		p := ipv6.NewPacketConn(conn)
```

Comment:
This seems weird to me - you can make a packet conn out of an existing UDP packet conn?

This doesn't do another layer of marshaling UDP packets on top of UDP packets, does it?

Reply:
I believe this just joins the multicast group and discards, I'll double-check in the code before merging, after the wall of meetings I have now :(

Reply:
(and the CI failures of course)

Reply:
oh the CI failure is for router7, I'll make a PR

Reply:
According to the documentation at https://godoc.org/golang.org/x/net/ipv6 it should be fine, looking at the code now

Reply:
seems just a wrapper, I think it's safe. I'll merge for now (we can unland if we find subtleties later)